src/main/java/seedu/guilttrip/logic/commands/remindercommands/AddGeneralReminderCommand.java

```diff
@@ -0,0 +1,123 @@
+package seedu.guilttrip.logic.commands.remindercommands;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.guilttrip.logic.parser.CliSyntax.PREFIX_DESC;
+import static seedu.guilttrip.logic.parser.CliSyntax.PREFIX_END_DATE;
+import static seedu.guilttrip.logic.parser.CliSyntax.PREFIX_LOWER_BOUND;
+import static seedu.guilttrip.logic.parser.CliSyntax.PREFIX_START_DATE;
+import static seedu.guilttrip.logic.parser.CliSyntax.PREFIX_TYPE;
+import static seedu.guilttrip.logic.parser.CliSyntax.PREFIX_UPPER_BOUND;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+
+import seedu.guilttrip.logic.CommandHistory;
+import seedu.guilttrip.logic.commands.Command;
+import seedu.guilttrip.logic.commands.CommandResult;
+import seedu.guilttrip.logic.commands.exceptions.CommandException;
+import seedu.guilttrip.model.Model;
+import seedu.guilttrip.model.entry.Date;
+import seedu.guilttrip.model.entry.Description;
+import seedu.guilttrip.model.reminders.GeneralReminder;
+import seedu.guilttrip.model.reminders.conditions.Condition;
+import seedu.guilttrip.model.reminders.conditions.DateCondition;
+import seedu.guilttrip.model.reminders.conditions.QuotaCondition;
+import seedu.guilttrip.model.reminders.conditions.TagsCondition;
+import seedu.guilttrip.model.reminders.conditions.TypeCondition;
+import seedu.guilttrip.model.tag.Tag;
+
+
+/**
+ * Adds a entry to the guilttrip book.
+ */
+public class AddGeneralReminderCommand extends Command {
+
+    public static final String COMMAND_WORD = "addReminder";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Adds a GeneralReminder to reminders list. "
```

Comment:
Same here. Tip: you can use <kbd>Ctrl</kbd>+<kbd>Alt</kbd>+<kbd>V</kbd> for abstracting the highlighted element into a variable.